TextFormatter.java

```diff
@@ -1,47 +1,129 @@
 class TextFormatter {
   private int maxLineLength;
 
-  private static final String text = "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy " +
-          "eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et " +
-          "accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem " +
-          "ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod " +
-          "tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et " +
-          "justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est " +
-          "Lorem ipsum dolor sit amet.";
+  private static final String text = "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy "
+      + "eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et "
+      + "accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem "
+      + "ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod "
+      + "tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et "
+      + "justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est "
+      + "Lorem ipsum dolor sit amet.";
 
   public static void main(String[] args) {
     TextFormatter formatter = new TextFormatter(30);
-    formatter.print(text);
+
+    formatter.print(text, "rightbound");
+    formatter.print(text, "leftbound");
+  }
+
+  private void print(String text, String formateType) {
+    if (formateType == "leftbound") {
+      formateLeftbound(text);
+    } else {
+      formateRightbound(text);
+    }
   }
 
   // Konstruktor
   public TextFormatter(int maxLineLength) {
     this.maxLineLength = maxLineLength;
   }
 
-  // Ausgabe
-  public void print(String aText) {
-   String formattedString = "";
+  private String fillWithBlanks(String line) {
+    int length = line.length();
+    for (int i = 0; i < (maxLineLength - length); i++) {
+      line = " " + line;
+    }
+    return line;
+  }
+
+  public void formateRightbound(String aText) {
```

Comment:
Sie haben es vermutlich bereits selbst gemerkt: 
Diese Methode ist furchtbar lang, sehr verschachtelt und damit nicht leicht zu begreifen. Überlegen Sie sich, ob Sie diese Methode in einem halben Jahr noch verstehen können? Deshalb zerlegen wir solch lange Methoden in Aufrufe mit sprechenden Namen (so wie Sie es ja bereits mit fillWithBlanks gemacht haben.